cpp_utils/BLEAdvertising.cpp

```diff
@@ -280,28 +279,28 @@ void BLEAdvertisementData::setAppearance(uint16_t appearance) {
  */
 void BLEAdvertisementData::setCompleteServices(BLEUUID uuid) {
 	char cdata[2];
-	switch(uuid.bitSize()) {
+	switch (uuid.bitSize()) {
 		case 16: {
 			// [Len] [0x02] [LL] [HH]
 			cdata[0] = 3;
 			cdata[1] = ESP_BLE_AD_TYPE_16SRV_CMPL;  // 0x03
-			addData(std::string(cdata, 2) + std::string((char *)&uuid.getNative()->uuid.uuid16,2));
+			addData(std::string(cdata, 2) + std::string((char*) &uuid.getNative()->uuid.uuid16, 2));
 			break;
 		}
 
 		case 32: {
 			// [Len] [0x04] [LL] [LL] [HH] [HH]
 			cdata[0] = 5;
 			cdata[1] = ESP_BLE_AD_TYPE_32SRV_CMPL;  // 0x05
-			addData(std::string(cdata, 2) + std::string((char *)&uuid.getNative()->uuid.uuid32,4));
+			addData(std::string(cdata, 2) + std::string((char*) &uuid.getNative()->uuid.uuid32, 4));
 			break;
 		}
 
 		case 128: {
 			// [Len] [0x04] [0] [1] ... [15]
 			cdata[0] = 17;
 			cdata[1] = ESP_BLE_AD_TYPE_128SRV_CMPL;  // 0x07
-			addData(std::string(cdata, 2) + std::string((char *)uuid.getNative()->uuid.uuid128,16));
```

Comment:
Noticed this. Was this intentional? Am I missing something?
`&uuid.getNative()->uuid.uuid128`
vs
`uuid.getNative()->uuid.uuid128`

Reply:
https://github.com/espressif/esp-idf/blob/a557e8cc14880717f242918a754cbc37d85e036a/components/bt/bluedroid/api/include/api/esp_bt_defs.h#L87-L92

Reply:
Is it an error in my code or the original?..

Reply:
In that union uuid16 and uui32 are type uint_t so you have to use & to get address, uint128 is an array so it is already pointer. By passing variable uuint128 from that union you are passing pointer to first element of array. It was correct in this code.